### DISCUSSION TITLE
Is it possible to return a pydantic response model and a file response at the same time

### DISCUSSION THREAD

**Discussion by PrathamSoni** [fastapi/fastapi#11120] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel


app = FastAPI()

class Output(BaseModel):
    metadata = BaseModel
    output_files = List[Files]



@app.post("/")
async def post_root():
    return Output(...)
```

### Description

I'm trying to return a list of files as well as some other key value metadata. Is there a way to do this in a single response body?

### Operating System

Linux, macOS

### FastAPI Version

0.109

### Pydantic Version

2.5.1

### Python Version

3.11.5

**Comment by jgould22** [comment, marked as answer]
You can base64 encode the value of the file and return it as part of your response model, you will have to sort that out in the client if you want to turn it back into binary.

Another option, if you dont have many values or they are starts you can return them in response headers.

**Comment by aanchlia** [comment]
If the file is not large, something like this should work


```python
@app.post("/")
async def post_root():
    output =  Output(...)

    file_name = "test.txt"
    headers = { "Content-Disposition": f"attachment; file_name={file_name}"}

    return JSONResponse(content=output, headers=headers)
```
